content/en/docs/reference/node/cri-api-development-policies.md

```diff
@@ -0,0 +1,107 @@
+---
+content_type: "reference"
+title: Kubernetes feature development and Container runtimes
+weight: 10
+---
+
+The mechanics of a feature development that requires new CRI APIs is covered in
+documentation on CRI API [feature-development](https://github.com/kubernetes/cri-api?tab=readme-ov-file#feature-development).
+This article declares policies for developing new Kubernetes features
+that require CRI API changes. The goal of these policies is to ensure great user
+experience for people trying the new feature early, adopting it when it is
+enabled by default, and relying on it as a GA functionality.
+
+## Supported container runtimes
+
+Features and CRI API are supposed to be portable and generic and not limited to
+a specific container runtime. However at this moment we require every feature to
+work on two container runtimes: [Containerd](https://containerd.io/) and
+[CRI-O](https://cri-o.io/). These are two runtimes that are tested as part of a
+kubernetes development and release process. When this document refers to two
+container runtimes, it assumes both - Containerd and CRI-O. If any other
+container runtimes begin working actively with the Kubernetes community, this
+document will need to be updated.
+
+## Same maturity level (for beta and GA)
+
+Implementation of an API needed for a Kubernetes feature in a container runtime
+MUST be at least the same maturity as in k8s at a moment of Kubernetes release.
+This is similar to the [deprecation policy](/docs/reference/using-api/deprecation-policy/#deprecating-a-feature-or-behavior)
+when one feature is replacing another.
+
+With Containerd and CRI-O it means that for the GA features, there should be a
+release of Containerd and CRI-O implementing APIs needed for the feature in
+those container runtimes. And those APIs MUST NOT be marked as experimental
+features ([Containerd experimental features](https://containerd.io/releases/#experimental-features)).
+For the beta, neither container runtime has a notion of beta feature or release and
+realistically the same maturity criteria applies as for GA.
+
+## Same maturity level (for alpha)
+
+There should be at least one implementation of an API needed for the Kubernetes
+feature merged into the container runtime default branch (main) or marked as
+experimental. An e2e test may demonstrate that the feature is working should be
+merged alongside the code. Note, tests may run against the default branch of a
+container runtime and the feature can be still not shipped.
+
+The actual container runtime release may be delayed to the later stage, but
+Kubernetes highly encourages fast availability of a release of a container
+runtime that can be tested by early adopters.
+
+## Minimal number of implementations
+
+Both Containerd and CRI-O MUST have a GA release with the implementations of an
+API needed for a Kubernetes feature before this Kubernetes feature can be
+promoted to GA.
```

Comment:
Avoid making this sound like a commitment. Imagine if the containerd project closed and everyone used to a rival thing (Docker maybe?)

We would still release Kubernetes and wouldn't require eg a bump to major version 2 to bypass this. It's _not_ a commitment.

Reply:
This is a commitment to the extend. If it will close we will rewrite the document. The goal to not let the end users down with the feature implemented with no reasonable way to use it on a supported runtime

Reply:
My opinion here: a commitment we rewrite when the context changes doesn't sound like an actual commitment.

I recommend getting SIG Architecture to make a note somewhere about what commitments we are definitely making, and which are a strong promise to make best efforts.